x-pack/filebeat/input/googlepubsub/input.go

```diff
@@ -69,6 +69,7 @@ func NewInput(
 	out, err := connector.ConnectWith(cfg, beat.ClientConfig{
 		Processing: beat.ProcessingConfig{
 			DynamicFields: inputContext.DynamicFields,
+			KeepNull:      conf.KeepNull,
```

Comment:
KeepNull will be set be automatically set by the `pipelineConnector` in `filebeat/channel/connector.go`. This makes me wonder if we need to set `Processing` at all.

Reply:
> KeepNull will be set be automatically set by the `pipelineConnector` in `filebeat/channel/connector.go`.

I understand now why `KeepNull` is not needed here — because it's already being set here: https://github.com/elastic/beats/pull/13928/files#diff-6c297ac7bc396fe6889d5668390af04eR109. Thanks.

> This makes me wonder if we need to set `Processing` at all.
I do think we need to set `Processing` here because of the `DynamicFields` config option. The `pipelineConnector` doesn't set it in its `ConnectWith` method.